cmd/apiserver_receive_adapter/main.go

```diff
@@ -40,13 +41,21 @@ var (
 	kubeconfig = flag.String("kubeconfig", "", "Path to a kubeconfig. Only required if out-of-cluster.")
 )
 
+type StringList []string
+
+func (s *StringList) Decode(value string) error {
```

Comment:
I assume you chose the custom decoder here because default splitter in envconfig (iuc) is comma which is a valid labelselector character? If so, consider commenting the reasoning in case it's not obvious.

Reply:
Total nit, but I'm wondering whether `;` would read better than `|`, since `|` at the end of line can introduce a [literal block scalar](https://yaml.org/spec/1.2/spec.html#id2795688).

Reply:
If I read correctly, the disallowed characters are: `,()!=`

Reply:
Actually, where is this used?

Reply:
Decode is used internally by envconfig. 

`|` is fine as I don't expect end-users to directly manipulate adapters. `;` works too and is maybe more conventional. Changing...

Reply:
Aha, I'd missed the envConfig connection, thanks!

I also missed that this was an internal detail, so both `|` and `;` seem reasonable, though potentially problematic in the sense that both have special shell meaning.